src/main/java/com/smartbear/readyapi/client/execution/CodegenBasedTestServerApi.java

```diff
@@ -296,8 +300,41 @@ public void setBasePath(String basePath) {
     }
 
     @Override
-    public ProjectResultReport postProject(ProjectExecutionRequest executionRequest, boolean async, HttpBasicAuth auth) throws ApiException {
+    public ProjectResultReport postSwagger(File swaggerFile, TestServerRequestExecutor.SwaggerFormat swaggerFormat,
```

Comment:
yes, this will be even better. But we still need to rename RecipeExcutor (since it executes project as well) - which I renamed already in this PR, OR should we have RecipeExecutor, ProjectExecutor, ApiValidtor  - and create them from TestServerClient as you have suggested above?